components/suggest/src/db.rs

```diff
@@ -271,23 +294,24 @@ impl<'a> SuggestDao<'a> {
         for suggestion in suggestions {
             self.scope.err_if_interrupted()?;
             let suggestion_id: i64 = self.conn.query_row_and_then_cachable(
-                "INSERT INTO suggestions(
-                     record_id,
-                     provider,
-                     title,
-                     url
-                 )
-                 VALUES(
-                     :record_id,
-                     :provider,
-                     :title,
-                     :url
-                 )
-                 RETURNING id
-                 ",
+                &format!(
+                    "INSERT INTO suggestions(
+                         record_id,
+                         provider,
+                         title,
+                         url
+                     )
+                     VALUES(
+                         :record_id,
+                         {},
+                         :title,
+                         :url
+                     )
+                     RETURNING id",
+                    SuggestionProvider::Amo as u8
```

Comment:
The `SuggestionProvider` variants are all integers, so, as with `providers_to_sql_list` above, we can safely interpolate them directly into the query string, without worrying about SQL injection. This is a trick we use in Places, too! ✨